### DISCUSSION TITLE
`useForm` errorState as dependency for `useMemo` not triggering

### DISCUSSION THREAD

**Discussion by bombillazo** [react-hook-form/react-hook-form#9933] [General]
Hello, we are seeing some unexpected behavior when using the `useForm` hook in conjunction with `useMemo`. We setup a from and want to get errors for usage in some external components. When we try to memoize the error though, the value is never updated:

```ts
// some functional component
() => {
  const form = useForm<F>({
    mode: 'all',
  });

  ...

  // value not changing!
  const formError = useMemo(
    () => form.formState.errors[formErrorPath]?.message as string,
    [form.formState.errors],
  );

  ...

  return (...);
}
```
We could assign the value directly to the variable but we want to leverage the memoization. Any ideas how to achieve this? Thans.

**Comment by Moshyfawn** [comment]
`formState` is a `Proxy` object. You need to [read the value first](https://react-hook-form.com/api/useform/formstate#rules) in order to subscribe to it.

**Reply by bombillazo** [reply]
Is reading as simple as destructuring or assigning it to a variable?
